app/helpers/s3.py

```diff
@@ -62,6 +63,14 @@ def get_s3_file(wmts_path, etag=None):
             settings.AWS_BUCKET_HOST, timeout=settings.HTTP_CLIENT_TIMEOUT
         )
         http_client.request("GET", path, headers=headers)
+        my_socket = http_client.sock
+        # Set the following socket options
+        # SO_KEEPALIVE: 1 => Enable TCP keepalive
+        # TCP_KEEPIDLE: 60 => Time in seconds until the first keepalive is sent
+        # TCP_KEEPINTVL: 60 => How often should the keepalive packet be sent
+        my_socket.setsockopt(socket.SOL_SOCKET, socket.SO_KEEPALIVE, 1)
+        my_socket.setsockopt(socket.SOL_TCP, socket.TCP_KEEPIDLE, 120)
+        my_socket.setsockopt(socket.SOL_TCP, socket.TCP_KEEPINTVL, 120)
```

Comment:
Should we not set the socket option before sending the request on line 65 ? Based on docu `http.client.HTTPConnection.request()` send the request.

Reply:
thank you @ltshb 
yes i was not aware of that. i will try to fix this.